apps/remote_control/lib/lexical/remote_control/project_node.ex

```diff
@@ -0,0 +1,210 @@
+      :"#{Project.name(project)}@127.0.0.1"
+    end
+
+    defp path_append_arguments(paths) do
+      Enum.flat_map(paths, fn path ->
+        ["-pa", Path.expand(path)]
+      end)
+    end
+
+    defp port_wrapper_executable do
+      :remote_control
+      |> :code.priv_dir()
+      |> Path.join("port_wrapper.sh")
+    end
+  end
+
+  alias Lexical.RemoteControl.ProjectNodeSupervisor
+  use GenServer
+
+  def start(project, project_listener, paths) do
+    node_name = State.node_name(project)
+    remote_control_config = Application.get_all_env(:remote_control)
+
+    {:ok, node_pid} = ProjectNodeSupervisor.start_project_node(project)
+
+    with :ok <- start_node(project, paths),
+         :ok <-
+           :rpc.call(node_name, RemoteControl.Bootstrap, :init, [
+             project,
+             project_listener,
+             remote_control_config
+           ]) do
+      {:ok, node_pid}
+    end
+  end
+
+  @stop_timeout 1_000
+
+  def stop(%Project{} = project, stop_timeout \\ @stop_timeout) do
+    project |> name() |> GenServer.call({:stop, stop_timeout}, stop_timeout + 100)
+  end
+
+  def child_spec(%Project{} = project) do
+    %{
+      id: name(project),
+      start: {__MODULE__, :start_link, [project]},
+      restart: :transient
+    }
+  end
+
+  def start_link(%Project{} = project) do
+    state = State.new(project)
+    GenServer.start_link(__MODULE__, state, name: name(project))
+  end
+
+  @start_timeout 3_000
+
+  defp start_node(project, paths) do
+    project |> name() |> GenServer.call({:start, paths}, @start_timeout + 500)
+  end
+
+  @impl GenServer
+  def init(state) do
+    Process.flag(:trap_exit, true)
+    {:ok, state}
+  end
+
+  @impl true
+  def handle_call({:start, paths}, from, %State{} = state) do
+    :ok = :net_kernel.monitor_nodes(true, node_type: :visible)
+    Process.send_after(self(), :maybe_start_timeout, @start_timeout)
+    state = State.start(state, paths, from)
+    {:noreply, state}
+  end
+
+  @impl true
+  def handle_call({:stop, stop_timeout}, from, %State{} = state) do
+    state = State.stop(state, from, stop_timeout)
+    {:noreply, state, stop_timeout}
+  end
+
+  @impl true
+  def handle_info({:nodeup, _node, _}, %State{} = state) do
+    GenServer.reply(state.started_by, :ok)
+    {pid, _ref} = state.started_by
+    Process.monitor(pid)
+    {:noreply, State.handle_nodeup(state)}
+  end
+
+  @impl true
+  def handle_info(:maybe_start_timeout, %State{status: :started} = state) do
+    {:noreply, state}
+  end
+
+  @impl true
+  def handle_info(:maybe_start_timeout, %State{} = state) do
+    GenServer.reply(state.started_by, {:error, :start_timeout})
+    {:stop, :start_timeout, nil}
+  end
+
+  @impl true
+  def handle_info({:nodedown, _, _}, %State{} = state) do
+    GenServer.reply(state.stopped_by, :ok)
+    state = State.on_nodedown(state)
+    {:stop, :shutdown, state}
+  end
+
+  @impl true
+  def handle_info({:DOWN, _ref, :process, _object, _reason}, %State{} = state) do
```

Comment:
However, I am inquisitive. Since `RemoteControl.ProjectNode` and `Server.Project.Node` have already been linked, why doesn't the death of the project node vm cause the death of the `Server.Project.Node` process? Normally, both sides need to use a flag(:trap_exit).